File: src/uaparser_ffi.erl
-module(uaparser_ffi).
-export([cache_get/1, cache_put/2]).

cache_get(Key) ->
    try persistent_term:get(Key) of
        Val -> {ok, Val}
    catch
        error:badarg -> {error, nil}
    end.

cache_put(Key, Val) ->
    persistent_term:put(Key, Val),
    nil.

File: src/uaparser_ffi.mjs
import { Result$Error, Result$Ok } from "./gleam.mjs";

const cache = new Map();

export function cache_get(key) {
  if (cache.has(key)) {
    return Result$Ok(cache.get(key));
  }
  return Result$Error(undefined);
}

export function cache_put(key, val) {
  cache.set(key, val);
  return undefined;
}
